### PR TITLE
fix: rm checksum for dependencies from local path and fix symlink

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [x] N
- [ ] Y 

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -754,6 +749,11 @@ func (c *KpmClient) VendorDeps(kclPkg *pkg.KclPkg) error {
 	return nil
 }
 
+func depExisted(localPath string, dep pkg.Dependency) bool {
```
Comment (Peefy) [inline pkg/client/client.go:752]: Add more comments here.

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -284,6 +284,44 @@ func DirExists(path string) bool {
 	return err == nil
 }
 
+// IsSymlinkValidAndExists will check whether the symlink exists and points to a valid target
+// return three values: whether the symlink exists, whether it points to a valid target, and any error encountered
+func IsSymlinkValidAndExists(symlinkPath string) (bool, bool, error) {
```
Comment (Peefy) [inline pkg/utils/utils.go:289]: Add unit test cases for the windows platform.

File: pkg/client/client.go (hunk 1 above)
Comment (zong-zhe) [reply pkg/client/client.go:755]: fixed

File: pkg/utils/utils.go (hunk 2 above)
Comment (zong-zhe) [reply pkg/utils/utils.go:290]: `IsSymlinkValidAndExists` is only used on unix-like system, I added some comments for this.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**